File: src/main/java/com/cjw/reactivecommunityproject/web/bbs/controller/BbsController.java
package com.cjw.reactivecommunityproject.web.bbs.controller;

import com.cjw.reactivecommunityproject.common.spring.model.response.RestResponseVO;
import com.cjw.reactivecommunityproject.web.bbs.model.request.BbsCreateVO;
import com.cjw.reactivecommunityproject.web.bbs.model.request.BbsModifyVO;
import com.cjw.reactivecommunityproject.web.bbs.service.BbsService;
import com.cjw.reactivecommunityproject.web.bbs.validation.BbsValidationGroup;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/rest/bbs")
public class BbsController {
    private final BbsService bbsService;

    @PostMapping
    public ResponseEntity<RestResponseVO<Void>> create(@RequestBody @Validated(BbsValidationGroup.Create.class) BbsCreateVO bbsContent) {
        return ResponseEntity.ok(bbsService.create(bbsContent));
    }


    @PatchMapping
    public ResponseEntity<RestResponseVO<Void>> modify(@RequestBody @Validated(BbsValidationGroup.Modify.class) BbsModifyVO bbsModifyVO) {
        return ResponseEntity.ok(bbsService.modify(bbsModifyVO));
    }


    @DeleteMapping("/{uid}")
    public ResponseEntity<RestResponseVO<Void>> remove(
            @PathVariable("uid") Long uid
    ) {
        return ResponseEntity.ok(bbsService.remove(uid));
    }
}

File: src/main/kotlin/com/devcjw/reactivecommunity/board/controller/BbsController.kt
package com.devcjw.reactivecommunity.board.controller

import com.devcjw.reactivecommunity.auth.model.domain.RcUserJwtClaims
import com.devcjw.reactivecommunity.board.model.domain.RepBbsListVO
import com.devcjw.reactivecommunity.board.service.BbsService
import com.devcjw.reactivecommunity.common.model.RestResponseVO
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import lombok.RequiredArgsConstructor
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux

@RestController
@RequestMapping("/bbs")
@RequiredArgsConstructor
@Tag(name = "게시판 컨트롤러", description = "게시판을 담당하는 컨트롤러")
class BbsController(
    private val bbsService: BbsService
) {
    @GetMapping
    @Operation(summary = "게시판 목록 불러오기", description = "게시판 목록을 불러오는 API")
    fun list(
        @AuthenticationPrincipal rcUser: RcUserJwtClaims
    ): Flux<RestResponseVO<RepBbsListVO>> {
        return bbsService.list(rcUser)
    }
}